packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -46,12 +46,20 @@ export const SelectionHandler = (
 
   const { store, selection } = state;
 
-  let currentTarget: TextAnnotationTarget | undefined;
+  let currentTarget: {
+    previous?: TextAnnotationTarget,
```

Comment:
Not ideal (and horrible naming). But we need to record the state of the annotation *before* this segment was added.